app/src/main/java/com/github/se/eduverse/repository/DashboardRepository.kt

```diff
@@ -34,15 +37,25 @@ class DashboardRepositoryImpl(private val firestore: FirebaseFirestore) : Dashbo
                 }
             awaitClose { listener.remove() }
           }
-          .catch { emit(emptyList()) } // Catch any exceptions and emit an empty list
+          .catch { emit(emptyList()) }
```

Comment:
You might want to log errors in your catch block to ensure you can debug issues effectively.